app/config/security.php

```diff
@@ -134,12 +129,25 @@
         ],
         'firewalls'      => $firewalls,
         'access_control' => [
-            ['path' => '^/api', 'roles' => 'IS_AUTHENTICATED_FULLY'],
+            // First there should be URIs for login or definitely public ones.
+            ['path' => '^/installer', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
+            ['path' => '^/(_(profiler|wdt)|css|images|js)/', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
+            ['path' => '^/s/login$', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
+            ['path' => '^/s/sso_login', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
+            ['path' => '^/s/saml/login$', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
+            ['path' => '^/saml/discovery$', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
+            ['path' => '^/oauth/v2/authorize', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
+            // Second should be URIs that are defined as non-public.
+            ['path' => '^/api', 'roles' => AuthenticatedVoter::IS_AUTHENTICATED_FULLY],
+            ['path' => '^/(s/|elfinder|efconnect)', 'roles' => AuthenticatedVoter::IS_AUTHENTICATED],
+            // Last the URIs that are none of the above.
+            ['path' => '^/', 'roles' => AuthenticatedVoter::PUBLIC_ACCESS],
         ],
     ]
 );
 
 $container->setParameter('mautic.saml_idp_entity_id', '%env(MAUTIC_SAML_ENTITY_ID)%');
+$container->setParameter('mautic.saml_enabled', '%env(MAUTIC_SAML_ENABLED)%');
```

Comment:
```suggestion
$container->setParameter('mautic.saml_enabled', '%env(bool:MAUTIC_SAML_ENABLED)%');
```
Let's use an environment variable processor.

Reply:
Done.